google/cloud/future_generic.h

```diff
@@ -60,6 +60,15 @@ class future final : private internal::future_base<T> {
   // NOLINTNEXTLINE(google-explicit-constructor)
   future(future<future<T>>&& rhs) noexcept(false);
 
+  /**
+   * Creates a future from a future whose result type is convertible to this
+   * future's result type.
+   */
+  template <class U,
+            typename Enable = std::enable_if<std::is_constructible<T, U>{}>>
+  future(future<U>&& rhs)
```

Comment:
This needs an `explicit`

Reply:
So, the purpose was for this to be an implicit conversion- such that it could operate on function return values. But I could make it explicit if you'd prefer

Reply:
My gut says we should not allow implicit conversions, I might be wrong, but we can always add them later, while removing them is a breaking change.  Please make these explicit

Reply:
ack

Reply:
Done.